lib/THC/THCTensorTopK.cu

```diff
@@ -261,7 +261,7 @@ __global__ void gatherTopK(TensorInfo<float, IndexType> input,
                            IndexType numTopKSlices,
                            IndexType topKWithinSliceStride,
 
-                           TensorInfo<float, IndexType> indices,
+                           TensorInfo<long, IndexType> indices,
```

Comment:
there's a lot more that needs changing in this function to use long indices, if you are calling top-k on a tensor with more than `2^31 - 1` elements FYI